x/oracle/types/currency_deviation_thresholds.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import (
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// String implements fmt.Stringer interface
+func (cdt CurrencyDeviationThreshold) String() string {
+	out, _ := yaml.Marshal(cdt)
+	return string(out)
+}
+
+func (cdt CurrencyDeviationThreshold) Equal(cdt2 *CurrencyDeviationThreshold) bool {
+	if !strings.EqualFold(cdt.BaseDenom, cdt2.BaseDenom) || !strings.EqualFold(cdt.Threshold, cdt2.Threshold) {
+		return false
+	}
+
+	return true
```

Comment:
nit: could be one line

`return strings.EqualFold(cdt.BaseDenom, cdt2.BaseDenom) && strings.EqualFold(cdt.Threshold, cdt2.Threshold)`